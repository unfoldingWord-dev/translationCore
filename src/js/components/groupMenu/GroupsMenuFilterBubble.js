import React from 'react';
import PropTypes from 'prop-types';
import { Glyphicon } from 'react-bootstrap';

class GroupsMenuFilterBubble extends React.Component {

  handleBubbleRemove() {
    this.props.setFilter(this.props.name, false);
  }

  render() {
    return (
      <span className="filter-bubble-wrapper">
        <span className="filter-bubble">
          <Glyphicon className='filter-remove' glyph='remove' onClick={this.handleBubbleRemove.bind(this)} />
          <span className="filter-text">{this.props.text}</span>
        </span>
      </span>
    );
  }
}

GroupsMenuFilterBubble.propTypes = {
  name: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  setFilter: PropTypes.func.isRequired
};

export default GroupsMenuFilterBubble;
