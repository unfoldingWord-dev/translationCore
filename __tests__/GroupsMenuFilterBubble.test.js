/* eslint-disable no-console */
/* eslint-env jest */
import React from 'react';
import GroupsMenuFilterBubble from '../src/js/components/groupMenu/GroupsMenuFilterBubble';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Tests for GroupsMenuFilterBubble', () => {
  test('Test GroupsMenuFilterBubble', () => {
    const props = {
      name: 'showMe',
      text: 'My filter',
      setFilter: jest.fn()
    };

    // when
    const wrapper = shallow(
      <GroupsMenuFilterBubble {...props} />
    );

    // then
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('Test GroupsMenuFilterBubble handleBubbleRemove()', () => {
    const setFilter = jest.fn();
    const props = {
      name: 'showMe',
      text: 'My filter',
      setFilter: setFilter
    };

    // when
    const wrapper = shallow(
      <GroupsMenuFilterBubble {...props} />
    );
    wrapper.find('.filter-remove').simulate('click');

    // then
    expect(setFilter).toHaveBeenCalledWith(props.name, false);
  });
});
