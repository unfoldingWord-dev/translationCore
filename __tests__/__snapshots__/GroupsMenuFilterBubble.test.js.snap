// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for GroupsMenuFilterBubble Test GroupsMenuFilterBubble 1`] = `
<span
  className="filter-bubble-wrapper"
>
  <span
    className="filter-bubble"
  >
    <Glyphicon
      bsClass="glyphicon"
      className="filter-remove"
      glyph="remove"
      onClick={[Function]}
    />
    <span
      className="filter-text"
    >
      My filter
    </span>
  </span>
</span>
`;
